lib/source_file.h

```diff
@@ -214,8 +214,16 @@ class IHasSourceInfo {
     virtual ~IHasSourceInfo() {}
 };
 
+/// SFINAE helper to check if given class has a `getSourceInfo` method.
+template <class, class = void>
+struct has_SourceInfo : std::false_type {};
```

Comment:
Nit: `hasSourceInfo`? Or all lowercase snake case?

Not clear what conventions to use here...

Reply:
We already had `has_SourceInfo` (similar to `has_toString`, etc.), however, previously we checked if it was derived from `IHasSourceInfo`, I checked it instead of checking Java-style interface to checking whether `T.getSourceInfo()` is well-formed as this is was ultimately is called and more clean IMO.